addons/google_gmail/models/google_gmail_mixin.py

```diff
@@ -143,10 +143,14 @@ def _generate_oauth2_string(self, user, refresh_token):
         :return: The SASL argument for the OAuth2 mechanism.
         """
         self.ensure_one()
+        # token life is around 1 hour, add 5s threshold to gives
+        # time to contact google servers and avoid clock synchronization
+        # problems
+        threshold_seconds = 5
```

Comment:
Can you specify that 5 seconds is the timeout for the request? Also would give it maybe a couple seconds of leeway to be safe.

Ideally would probably make it some named constant so we don't have to think to update it here.
e.g. `GMAIL_TOKEN_REQUEST_TIMEOUT = 5` at the top of the file